acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/steps/site/SiteSteps.java

```diff
@@ -199,6 +200,38 @@ public void thenIShouldSeeTheFollowing(DataTable contentsTable) throws Throwable
         }
     }
 
+    @Then("^I (?:can|should) see elements with attributes:")
+    public void thenIShouldSeeMultipleElementsWithAttributes(final DataTable elementAttributes) throws Throwable {
+        final List<List<String>> rawElementItems = elementAttributes.raw();
+        List<String> keys = rawElementItems.get(0);
+        for (int i = 1; i < rawElementItems.size(); i++) {
+            List<String> elementItem = rawElementItems.get(i);
+
+            String xPathExpression = buildXPathExpression(keys, elementItem);
+
+            assertNotNull(
+                "I should find element with xpath expression " + xPathExpression,
+                sitePage.findElementWithXPath(xPathExpression)
+            );
+        }
+    }
+
+    private String buildXPathExpression(List<String> keys, List<String> elementItem) {
```

Comment:
The method name could probably be more specific, reflecting some details of the expression that it builds.

Not that I can think of any good name myself, mind you :grin:

Reply:
`buildXPathExpressionFromElementAttributes`? `buildXPathExpressionForElement`?

Reply:
`buildXPathExpressionFromElementAttributes` probably sounds better of the two :+1: